vcf.c

```diff
@@ -819,13 +829,17 @@ int bcf_hdr_parse(bcf_hdr_t *hdr, char *htxt)
             }
             p += len;
         }
-        if (hrec == NULL && len != 0) {
-            char *eol = strchr(p, '\n');
-            if (*p != '\0') {
-                hts_log_warning("Could not parse header line: %.*s",
-                                eol ? (int)(eol - p) : INT_MAX, p);
-            }
+        assert(hrec == NULL);
+        if (len < 0) {
+            // len < 0 indicates out-of-memory, or similar error
+            hts_log_error("Could not parse header line: %s", strerror(errno));
```

Comment:
Do we no longer wish to warn on the contents of the line that we couldn't parse?  Admittedly the `eol ? (int)(eol - p) : INT_MAX` code is a bit of a bizarrity in style, but that came from b8c6f7dd5581f4b7fcf0a997bfa80763a7ee9026.

I think it's still useful to be able to see the offending line and don't see why we'd want to hide that.  Admittedly it'll now only ever be the first line instead of every single line, but that's fair enough and usually the first one is sufficient.

Reply:
In this case (`len < 0`) the failure would be due to something like out-of-memory, so my reasoning was that the contents of the header line aren't likely to be relevant - unless it happens to be incredibly long.  All other types of dodgy line should now be printed out once only, with unprintable characters escaped and a reasonable cap on the length.

Reply:
Ah yes, I see they're still printed elsewhere, and if this is just malloc error then fair enough.  strerror will be more useful (provided errno has indeed been set, which it will if malloc failure, and isn't a left-over remnant of something else).

I'm not sure 320 chars is long enough for some valid VCF header lines though (but INT_MAX was a tad over the top!).

However checking a few local files the longest pre-CHROM line I see is 311 (so pretty close), and the longest CHROM is obviously unlimited, with ~10Kb being the longest in the files I have locally.

Reply:
Most of the VCF header lines I've seen are fairly short.  What was in the 311 character long one?

The `#CHROM` line only gets a very basic test in `bcf_hdr_parse()` itself, with most checks delegated to `bcf_hdr_parse_sample_line()` later.  If the basic test fails, you've likely either started a header with `#` instead of `##`; spelled `CHROM` or `POS` incorrectly; or forgotten the `#CHROM` line completely.  Either way a small amount of context should be enough to see where the problem is.

Reply:
It was a `##INFO` tag with a length comment:

```
##INFO=<ID=CIGAR,Number=A,Type=String,Description="The extended CIGAR representation of each alternate allele, with the exception that '=' is replaced by 'M' to ease VCF parsing.  Note that INDEL alleles do not have the first matched base (which is provided by default, per the spec) referred to by the CIGAR.">
```

Because descriptions can be any length, it's hard to know what's a sensible limit.

Reply:
Yes, I think it might be considered a feature that the warning messages get a length cap.